### PR TITLE
My recent improvements to the backport script.

### PR DESCRIPTION
**Pull request by Zimmi48**
Add new options `--no-conflict` and `--no-signature-check`, refactor the handling of options.

@maximedenes is the secondary code owner for that file but I wouldn't mind having a quick review from a bash knowledgeable person (like @SkySkimmer).

### CODE REVIEW

File: dev/tools/backport-pr.sh [hunk 1]
```diff
@@ -1,11 +1,31 @@
...
+        --no-signature-check)
+            NO_SIGNATURE_CHECK="true"
+            shift
+            ;;
+        --stop-before-merging)
+            STOP_BEFORE_MERGING="true"
+            shift
+            ;;
+        *)
+            PRNUM="$1"
```
Comment (SkySkimmer) [inline dev/tools/backport-pr.sh]: Maybe check that PRNUM isn't already set (don't silently ignore multiple PRNUMs)

File: dev/tools/backport-pr.sh [hunk 2]
```diff
@@ -30,6 +50,14 @@ MESSAGE=$(git log master --grep "Merge PR #${PRNUM}" --format="%s" | sed 's/Merg
 if git checkout -b "${BRANCH}"; then
 
     if ! git cherry-pick -x "${RANGE}"; then
+        if [ "$NO_CONFLICTS" == "true" ]; then
```
Comment (SkySkimmer) [inline dev/tools/backport-pr.sh]: Use [[ or single =

**Comment by Zimmi48** [comment]
Thanks @SkySkimmer!

File: dev/tools/backport-pr.sh [hunk 3]
```diff
@@ -30,6 +53,14 @@ MESSAGE=$(git log master --grep "Merge PR #${PRNUM}" --format="%s" | sed 's/Merg
 if git checkout -b "${BRANCH}"; then
 
     if ! git cherry-pick -x "${RANGE}"; then
+        if [[ "$NO_CONFLICTS" == "true" ]]; then
+            git status
+            echo "Conflicts! Aborting..."
+            git cherry-pick --abort
+            git checkout -
```
Comment (SkySkimmer) [inline dev/tools/backport-pr.sh]: Shouldn't this be `checkout --`?

Comment (SkySkimmer) [reply dev/tools/backport-pr.sh]: with -f too maybe
Test it manually until it works I guess.

Comment (Zimmi48) [reply dev/tools/backport-pr.sh]: It was tested manually ;-)
`git checkout -` is supposed to checkout back to the previous ref. AFAIK `git checkout --` (possibly with `-f`) would rather be about undoing partial changes, but this is already being taken care of by `git cherry-pick --abort`.

**Comment by Zimmi48** [comment]
Closing / reopening to test @coqbot.

**Comment by Zimmi48** [comment]
Yay! It works.

**Review by maximedenes — APPROVED** [review]

**Comment by maximedenes** [comment]
@SkySkimmer does this look good to you?

**Comment by SkySkimmer** [comment]
Sure.

**Pull request merged**